docs/metrics.md

```diff
@@ -0,0 +1,587 @@
+# OpenTelemetry Rust Metrics
+
+<details>
+<summary>Table of Contents</summary>
+
+* [Best Practices](#best-practices)
+* [Metrics API](#metrics-api)
+  * [Meter](#meter)
+  * [Instruments](#instruments)
+  * [Reporting measurements - use array slices for
+    attributes](#reporting-measurements---use-array-slices-for-attributes)
+  * [Reporting measurements via synchronous
+    instruments](#reporting-measurements-via-synchronous-instruments)
+  * [Reporting measurements via asynchronous
+    instruments](#reporting-measurements-via-asynchronous-instruments)
+* [MeterProvider Management](#meterprovider-management)
+* [Memory Management](#memory-management)
+  * [Example](#example)
+  * [Pre-Aggregation](#pre-aggregation)
+    * [Pre-Aggregation Benefits](#pre-aggregation-benefits)
+  * [Cardinality Limits](#cardinality-limits)
+    * [Cardinality Limits - Implications](#cardinality-limits---implications)
+    * [Cardinality Limits - Example](#cardinality-limits---example)
+  * [Memory Preallocation](#memory-preallocation)
+* [Metrics Correlation](#metrics-correlation)
+* [Modelling Metric Attributes](#modelling-metric-attributes)
+
+</details>
+
+## Best Practices
+
+// TODO: Add link to the examples, once they are modified to show best
+practices.
+
+## Metrics API
+
+### Meter
+
+:stop_sign: You should avoid creating duplicate
+[`Meter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Meter.html)
+instances with the same name. `Meter` is fairly expensive and meant to be reused
+throughout the application. For most applications, a `Meter` should be obtained
+from `global` and saved for re-use.
+
+The fully qualified module name might be a good option for the Meter name.
+Optionally, one may create a meter with version, schema_url, and additional
+meter-level attributes as well. Both approaches are demonstrated below.
+
+```rust
+use opentelemetry::global;
+use opentelemetry::InstrumentationScope;
+use opentelemetry::KeyValue;
+
+let scope = InstrumentationScope::builder("my_company.my_product.my_library")
+        .with_version("0.17")
+        .with_schema_url("https://opentelemetry.io/schema/1.2.0")
+        .with_attributes([KeyValue::new("key", "value")])
+        .build();
+
+// creating Meter with InstrumentationScope, comprising of
+// name, version, schema and attributes.
+let meter = global::meter_with_scope(scope);
+
+// creating Meter with just name
+let meter = global::meter("my_company.my_product.my_library");
+```
+
+### Instruments
+
+OpenTelemetry defines several types of metric instruments, each optimized for
+specific usage patterns. The following table maps OpenTelemetry Specification
+instruments to their corresponding Rust SDK types.
+
+:heavy_check_mark: You should understand and pick the right instrument type.
+
+> [!NOTE] Picking the right instrument type for your use case is crucial to
```

Comment:
This doesn't render right but i'm not sure if it's 1/ a syntax issue or 2/ something to do with the context in which it is rendered? I believe `[!NOTE]` and friends are part of github-flavoured markdown.

Reply:
https://github.com/github/docs/blob/main/content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md#alerts its supported in Github, will check why its not rendering.